src/ucp/core/ucp_worker.c

```diff
@@ -1679,11 +1680,13 @@ static ucs_status_t ucp_worker_init_mpools(ucp_worker_h worker)
     /* Create a hashtable of memory pools for mem_type devices */
     kh_init_inplace(ucp_worker_mpool_hash, &worker->mpool_hash);
 
+    UCS_MPOOL_PARAMS_INIT(mp_params, 0,
+                          sizeof(ucp_request_t) + context->config.request.size,
+                          0, UCS_SYS_CACHE_LINE_SIZE, 128, UINT_MAX, 0,
+                          1.0, &ucp_request_mpool_ops, "ucp_requests");
```

Comment:
this macro is not very nice; better write a function like ucs_mpool_params_reset() which sets default values for the parameters, and then every place that calls ucs_mpool_init should initialize the non-default parameters by setting every field explicitly.

Reply:
changed accordingly